code/game/machinery/vending.dm

```diff
@@ -464,20 +503,40 @@
 		if(src.coin)
 			display_records += src.coin_records
 
-		for (var/datum/data/vending_product/R in display_records)
-
-			// AUTOFIXED BY fix_string_idiocy.py
-			// C:\Users\Rob\Documents\Projects\vgstation13\code\game\machinery\vending.dm:285: dat += "<FONT color = '[R.display_color]'><B>[R.product_name]</B>:"
-			dat += {"<FONT color = '[R.display_color]'><B>[R.product_name]</B>:
-				<b>[R.amount]</b> </font>"}
-			// END AUTOFIX
-			if(R.price)
-				dat += " <b>($[R.price])</b>"
-			if (R.amount > 0)
-				var/idx=GetProductIndex(R)
-				dat += " <a href='byond://?src=\ref[src];vend=[idx];cat=[R.category]'>(Vend)</A>"
+		if(display_records.len > 12)
+			vertical = min(400 + (16 * (display_records.len - 12)),840)
+
+		categories["default"] = list()
+		var/list/category_names = list()
+		for (var/datum/data/vending_product/R in product_records)
+			if(R.subcategory)
+				if(!(R.subcategory in category_names))
+					category_names += R.subcategory
+					categories[R.subcategory] = list()
+				categories[R.subcategory] += R
 			else
-				dat += " <font color = 'red'>SOLD OUT</font>"
+				categories["default"] += R
+
+		for (var/datum/data/vending_product/R in categories["default"])
+			dat += GetProductLine(R)

```

Comment:
don't do this convert this stuff to docstrings {} instead of adding dat += multiple times strings are slow in BYOND.